client/js/components/navbar.jsx

```diff
@@ -23,7 +35,12 @@ class Navbar extends React.Component {
   }
 
   render() {
-    let iconElementRight = (<RaisedButton style={{marginTop: '6px'}} onClick={ () => this.refs.dialog.open()} label="Add new snippet" primary={true}/>);
+    let iconElementRight = '';
+    if ( this.state.currentUser ){
+      iconElementRight = (<RaisedButton style={{marginTop: '6px'}} onClick={ () => this.refs.dialog.open()} label="Add new snippet" primary={true}/>);
```

Comment:
you don't have to create anonymous function here. Just pass function. Try `onClick={ this.refs.dialog.open }`